bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +65,209 @@ def check_flax_supported() -> None:  # pragma: no cover
-        tag (:code:`Union[str, Tag]`):
-            Tag of a saved model in BentoML local modelstore.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
-        from_tf (:code:`bool`, `optional`, defaults to :code:`False`):
-            Load the model weights from a TensorFlow checkpoint save file.
-        from_flax (:code:`bool`, `optional`, defaults to :code:`False`):
-            Load the model weights from a Flax checkpoint save file
-        return_config (:code:`bool`, `optional`, default to :code:`False`):
-            Whether or not to return configuration of the Transformers model.
-        config_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`Config` object.
-        model_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`Model` object.
-        tokenizer_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`Tokenizer` object.
-        feature_extractor_kwargs (:code:`Dict[str, Any]`, `optional`):
-            Kwargs to pass into :code:`FeatureExtractor` object.
-        kwargs (:code:`Dict[str, Any]`, `optional`):
-            Other kwargs that can be parsed to transformers that is neither configs, model, tokenizer, and feature extractor.
-
-        .. warnings::
-            Make sure to add the corresponding kwargs for your Transformers :code:`Model`, :code:`Tokenizer`, :code:`Config`, :code:`FeatureExtractor` to the correct kwargs dict.
-
-        .. warnings::
-            Currently :code:`kwargs` accepts all kwargs for corresponding Pipeline.
-
-    Returns:
-        :obj:`Union[Pipeline, Tuple[Optional[PretrainedConfig], Union[PreTrainedModel, TFPreTrainedModel, FlaxPreTrainedModel], Optional[Union[PreTrainedTokenizer, PreTrainedTokenizerFast, PreTrainedFeatureExtractor]]]]`: either returning a
-        pipeline or a tuple containing :obj:`PretrainedConfig`, :obj:`Model` class object defined by :obj:`transformers`, with an optional :obj:`Tokenizer` class, or :obj:`FeatureExtractor` class for the given model saved in BentoML modelstore.
-
-    Examples:
+try:
+    import transformers
 
-    .. code-block:: python
+    _check_flax_supported()
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        """\
+        transformers is required in order to use module `bentoml.transformers`.
+        Instruction: Install transformers with `pip install transformers`.
+        """
+    )
 
-        import bentoml
-        model, tokenizer = bentoml.transformers.load('custom_gpt2')
 
-    If you want to returns an config object:
+class TransformersOptions(ModelOptions):
+    """Options for the Transformers model."""
 
-    .. code-block:: python
+    def __init__(self, **kwargs):
+        class _Schema(pydantic.BaseModel):
+            task: str
+            pipeline: bool = True
```

Comment:
It would also be nice to know about the Model, Tokenizer, Feature Extractor if any that is used in this Pipeline

Reply:
What do you suggest that we can do with that information?